securedrop/management/sources.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+from typing import List
+
+from db import db
+from encryption import EncryptionManager, GpgKeyNotFoundError
+from management import app_context
+from models import Source
+
+
+def remove_pending_sources(args: argparse.Namespace) -> int:
+    """
+    Removes pending source accounts, with the option of keeping
+    the n newest source accounts.
+    """
+    n = args.keep_most_recent
+    sources = find_pending_sources(n)
+    print(f"Found {len(sources)} pending sources")
+
+    deleted = []
+    for source in sources:
+        try:
+            EncryptionManager.get_default().delete_source_key_pair(source.filesystem_id)
+        except GpgKeyNotFoundError:
+            pass
+        delete_pending_source(source)
+        deleted.append(source)
+
+    print(f"Deleted {len(sources)} pending sources")
+    return 0
+
+
+def find_pending_sources(keep_most_recent: int) -> List[Source]:
+    """
+    Finds all sources that are marked as pending
+    """
+    with app_context():
+        pending_sources = (
```

Comment:
nit: you can return directly inside a `with` block and it does the right thing, you don't need to set it to a variable and return outside of it.